# CS 457 - Homework Assignment 4: Statistical Inference
## Name: Owais Waheed (ow07611)
### Email: ow07611@st.habib.edu.pk




#### Importing required libraries and reading data from csv file

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

df = pd.read_csv('EmployeeAttrition.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1470 entries, 0 to 1469
Data columns (total 32 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   Age                       1470 non-null   int64 
 1   BusinessTravel            1470 non-null   object
 2   DailyRate                 1470 non-null   int64 
 3   Department                1470 non-null   object
 4   DistanceFromHome          1470 non-null   int64 
 5   Education                 1470 non-null   int64 
 6   EducationField            1470 non-null   object
 7   EnvironmentSatisfaction   1470 non-null   int64 
 8   Gender                    1470 non-null   object
 9   HourlyRate                1470 non-null   int64 
 10  JobInvolvement            1470 non-null   int64 
 11  JobLevel                  1470 non-null   int64 
 12  JobRole                   1470 non-null   object
 13  JobSatisfaction           1470 non-null   int64 
 14  MaritalStatus           

### Q1. If the MonthlyIncome of Males is greater than Females


H0 -> Monthly income of males is (not greater) lesser than or equal to female.<br>
H1 -> Monthly income of males is greater than female <br>



In [2]:
male_s = df[(df['Gender'] == 'Male')]['MonthlyIncome']

female_s = df[(df['Gender'] == 'Female')]['MonthlyIncome']


stats.ttest_ind(male_s, female_s, alternative='greater')

Ttest_indResult(statistic=-1.2212617308870655, pvalue=0.8889084827245606)

Since, p_value = 0.888 > 0.05 we can not reject null hypothesis that is monthly income of males is less than female. And so we have to reject accepting hypothesis.
<br> Based on this test, we can say we do not have enough evidence to prove our accept hypothesis that monthly income of male is greater than females

### Q2. If the WorkLifeBalance of Males is less than Females

H0 ->  The Worklife Balance of males is not less (greatar or equal to) than females <br>
H1 -> The Worklife Balance of males is less than females.


In [3]:
male_wlb = df[(df['Gender'] == 'Male')]['WorkLifeBalance']
# print(male_wlb.sum()/len(male_wlb))
female_wlb = df[(df['Gender'] == 'Female')]['WorkLifeBalance']
print(female_wlb.sum()/len(female_wlb))
stats.ttest_ind(male_wlb, female_wlb, alternative='less')

2.7636054421768708


Ttest_indResult(statistic=-0.10546787956064164, pvalue=0.45800948246831286)

Since, p_value = 0.45 > 0.05 we can not reject null hypothesis that is worklife balance is not less than females.. And so we have to reject accepting hypothesis.
<br> Based on this test, we can say we do not have enough evidence to prove our accept hypothesis that worklife balance of males is less than females. 

### Q3. If the YearsAtCompany of Single is less than Married

H0 -> The Years at Company of single is not less (greater or equal to) than Married. <br>
H1 -> The Years at Company of single is less than Married.


In [4]:
single_yrs = df[(df['MaritalStatus'] == 'Single')]['YearsAtCompany']
# print(single_yrs.sum()/len(single_yrs))
married_yrs = df[(df['MaritalStatus'] == 'Married')]['YearsAtCompany']
# print(married_yrs.sum()/len(married_yrs))

stats.ttest_ind(single_yrs, married_yrs, alternative='less')

Ttest_indResult(statistic=-2.5589985737464476, pvalue=0.0053127663549315815)

Since, p_value = 0.0053 which is < 0.05 , we can reject the null hypothesis and safely conclude that singles have less years at companies than married employees.

### Q4. If the EnvironmentalSatisfaction of Attrition=Yes is less than Attrition=No

H0 -> The EnvironmentalSatisfaction of Attrition=Yes is not less (greater or equal) than EnvironmentalSatisfaction of Attrition=No. <br>
H1 -> The EnvironmentalSatisfaction of Attrition=Yes is less than EnvironmentalSatisfaction of Attrition=No.

In [5]:
yes = df[(df['Attrition']=='Yes') ]['EnvironmentSatisfaction']
no = df[(df['Attrition']=='No') ]['EnvironmentSatisfaction']

stats.ttest_ind(yes, no, alternative = 'less')

Ttest_indResult(statistic=-3.9818600026818496, pvalue=3.58616927468443e-05)

Since, p-value = 3.35e-05 which is way less than 0.05, we can reject the null hypothesis and can safely conclude that employees whose attrition = yes have less environment satisfaction and whose attrition = no have higher environment satisfaction. <br> 
So we might think that work environment can be one reason for attrition. But this statement requires more testing because this statistic test only shows corelation. so we can not drive causation solely based on this.

### Q5. If the MonthlyIncome of Manager is greater than Laboratory Technician (Hint: Use JobRole to find Manager and Laboratory Technician)

H0 -> The Monthly Income of Manager is not greater than (less or equal) MonthlyIncome of Laboratory Technician. <br>
H1 -> The Monthly Income of Manager is greater than MonthlyIncome of Laboratory Technician.

In [6]:
Manager_s = df[(df['JobRole'] == 'Manager')]['MonthlyIncome']
Lab_s = df[(df['JobRole'] == 'Laboratory Technician')]['MonthlyIncome']

stats.ttest_ind(Manager_s, Lab_s, alternative='greater')

Ttest_indResult(statistic=76.03317414634023, pvalue=9.971168334368433e-224)

Since p-value = 9.97e-224 which is way way lesser than 0.05, we can reject the null hypothesis and safely conclude that the monthly income of manager is greater than monthly income of Lab technician.  

### Q6. Is there any association between the Department and Gender?

H0 -> There is no association between the variables <br>
H1 -> There is association between the variables

In [7]:
chisqt = pd.crosstab(df['Gender'],df['Department'], margins=True)
value = np.array([chisqt.iloc[0][0:].values, chisqt.iloc[1][0:].values])
stats.chi2_contingency(value)



Chi2ContingencyResult(statistic=2.9644916359463056, pvalue=0.3971323304462201, dof=3, expected_freq=array([[ 25.2, 384.4, 178.4, 588. ],
       [ 37.8, 576.6, 267.6, 882. ]]))

From above, 0.397 is the p-value. As the p-value is greater than 0.05, we accept the NULL hypothesis and conclude that the variables ‘Department’ and ‘Gender’ are independent of each other and have no significant relationship.